junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestExtension.java

```diff
@@ -82,6 +90,7 @@ public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContex
 				.map(provider -> AnnotationConsumerInitializer.initialize(methodContext.method, provider))
 				.flatMap(provider -> arguments(provider, extensionContext))
 				.map(arguments -> {
+					validateArgumentCount(extensionContext, arguments);
```

Comment:
Playing around with this made me realize that we need to defer this validation until the invocation is being executed. Otherwise, all invocations will fail rather than just the offending ones. For example, annotating the added test in `ParameterizedTestDemo` with `@CsvSource({ "42, -666", "1", "2", "3" })` should result in the first invocation being reported as failed due to the new validation but all others should pass.

I think the cleanest way to do this is to return another extension from `ParameterizedTestInvocationContext#getAdditionalExtensions`, say `ArgumentCountValidator`, that implements `InvocationInterceptor` and overrides `interceptTestTemplateMethod` with the validation before calling `invocation.proceed()`.

Please let me know whether that's enough information for you to proceed. Alternatively, I'd also be willing to take over if you'd prefer that.

Reply:
Yes, I also thought something wasn't quite right after playing around with ParameterizedTestDemo. Thanks for the pointers with `ParameterizedTestInvocationContext#getAdditionalExtensions`. That definitely seems doable to me.

Reply:
I've updated the code, I think the Windows check is failing for unrelated reasons